client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -103,24 +104,146 @@ export const siteSetupFlow: Flow = {
+				switch ( currentStep ) {
+					case 'options':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 4 } } );
+						break;
+					case 'bloggerStartingPoint':
+						updateFlowProgress( { middleProgress: { progress: 2, count: 4 } } );
+						break;
+					case 'courses':
+					case 'designSetup':
+						updateFlowProgress( { middleProgress: { progress: 3, count: 4 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
+			}
+			case SiteIntent.Build: {
+				switch ( currentStep ) {
+					case 'designSetup':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 2 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
+			}
+			case SiteIntent.Sell: {
+				switch ( currentStep ) {
+					case 'options':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 6 } } );
+						break;
+					case 'storeFeatures':
+						updateFlowProgress( { middleProgress: { progress: 2, count: 6 } } );
+						break;
+				}
+				if ( storeType === 'simple' ) {
+					switch ( currentStep ) {
+						case 'designSetup':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				} else if ( storeType === 'power' ) {
+					switch ( currentStep ) {
+						case 'storeAddress':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'businessInfo':
+							updateFlowProgress( { middleProgress: { progress: 4, count: 6 } } );
+							break;
+						case 'wooConfirm':
+						case 'wooInstallPlugins':
+							updateFlowProgress( { middleProgress: { progress: 5, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				}
+
+				break;
+			}
+			case SiteIntent.Import: {
+				switch ( currentStep ) {
+					case 'import':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 4 } } );
+						break;
+					case 'importList':
+					case 'importReady':
+					case 'importReadyNot':
+					case 'importReadyWpcom':
+					case 'importReadyPreview': {
+						updateFlowProgress( { middleProgress: { progress: 2, count: 4 } } );
+						break;
+					}
+					case 'importerWix':
+					case 'importerBlogger':
+					case 'importerMedium':
+					case 'importerSquarespace':
+					case 'importerWordpress':
+						updateFlowProgress( { middleProgress: { progress: 3, count: 4 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
 			}
-		} else {
-			setStepProgress( undefined );
+		}
+
+		function updateFlowProgress( flowProgress: {
+			beginningStep?: string;
+			middleProgress?: { progress: number; count: number };
+			endStep?: string;
```

Comment:
Maybe we just need a boolean flag to indicate it's the final step

Reply:
I thought endStep would be similar to beginningStep (which consists different steps), but I think you're right, for now we only need a boolean